nxdlTypes.xsd

```diff
@@ -69,8 +69,27 @@
 			<xs:enumeration value="NX_TEMPERATURE"><xs:annotation><xs:documentation>example: K</xs:documentation></xs:annotation></xs:enumeration>
 			<xs:enumeration value="NX_TIME"><xs:annotation><xs:documentation>example: s</xs:documentation></xs:annotation></xs:enumeration>
 		    <xs:enumeration value="NX_TIME_OF_FLIGHT"><xs:annotation><xs:documentation>(alias to NX_TIME) example: s</xs:documentation></xs:annotation></xs:enumeration>
-		    <xs:enumeration value="NX_VOLTAGE"><xs:annotation><xs:documentation>example: V</xs:documentation></xs:annotation></xs:enumeration>
-		    <xs:enumeration value="NX_VOLUME"><xs:annotation><xs:documentation>example: m3</xs:documentation></xs:annotation></xs:enumeration>
+			<xs:enumeration value="NX_VOLTAGE"><xs:annotation><xs:documentation>example: V</xs:documentation></xs:annotation></xs:enumeration>
+			<xs:enumeration value="NX_TRANSFORMATION">
+				<xs:annotation>
+					<xs:documentation>
+						could be any of these: ``NX_LENGTH``, ``NX_ANGLE``, or ``NX_UNITLESS``
+						
+						There will be one or more transformations defined by one or more fields
+						for each transformation.  The units type ``NX_TRANSFORMATION`` designates the
+						particular axis generating a transformation (e.g. a rotation axis or a translation
+						axis or a general axis).   ``NX_TRANSFORMATION`` designates the
+						units will be appropriate to the type of transformation,
+						indicated in the :ref:`NXtransformations` base class
+						by the ``transformation_type`` value:
+						
+						* ``NX_LENGTH`` for ``translation``
+						* ``NX_ANGLE`` for ``rotation`` 
+						* ``NX_UNITLESS`` for axes for which no transformation type is specified.
+					</xs:documentation>
```

Comment:
BUT, the `anyUnitsAttr` is defined as `<xs:restriction base="xs:string">` with `<xs:enumeration value=...`.  These enumerations have no capability of `<xs:union` content.  The refactor of `anyUnitsAttr` from `xs:restriction` to `<xs:union` could have other consequences.

That, in itself using only the existing schema, is worth exploring in its own branch.